src/config.rs

```diff
@@ -149,28 +150,32 @@ fn read_devices(root: &Path, memtotal_mb: u64) -> Result<HashMap<String, Device>
 }
 
 fn locate_fragments(root: &Path) -> BTreeMap<String, PathBuf> {
-    let base_dirs = vec![
-        String::from(root.join("usr/lib").to_str().unwrap()),
-        String::from(root.join("usr/local/lib").to_str().unwrap()),
-        String::from(root.join("etc").to_str().unwrap()),
-        String::from(root.join("run").to_str().unwrap()), // We look at /run to allow temporary overriding
-                                                          // of configuration. There is no expectation of
-                                                          // programatic creation of config there.
+    let mut base_dirs = [
+        root.join("usr/lib"),
+        root.join("usr/local/lib"),
+        root.join("etc"),
+        root.join("run"), // We look at /run to allow temporary overriding
+                          // of configuration. There is no expectation of
+                          // programatic creation of config there.
     ];
 
     let cfg = FragmentScanner::new(
-        base_dirs.clone(),
+        base_dirs
+            .iter()
+            .cloned()
+            .map(|pb| pb.into_os_string().into_string().unwrap())
+            .collect(),
```

Comment:
I'm not convinced that this a net gain. The original code was relatively simple, and was using immutable data structures. I don't think saving a few memory allocations is important enough to add such complicated code and use mutable variables.

Reply:
Fair enough. I've since opened https://github.com/overdrop/liboverdrop-rs/pull/11, which will make most of the additional hard-type complexity here redundant. I'll revisit this if/when that gets merged, but I'll close it for now.